### PR TITLE
ui: statements page

### LINKED ISSUE #42154: ui: statement search & pagination follow up work
**Feedback for statement search #41537**

- [ ] Only underline statement link on hover
Screen Shot 2019-10-17 at 6 06 13 PM

- [ ] Remove statement modal that triggers from the bottom of the screen when clicking a statement in the list. Clicking a statement link from the list will take users directly to the statement detail page.
Screen Shot 2020-01-08 at 4 27 58 PM

- [x] Remove the previous/next tool tip OR remove the system tool tip. Sometimes they both pop up and overlap 
Screen Shot 2019-10-17 at 5 12 42 PM

- [x] If 0 results from search, hide pagination 
Screen Shot 2019-10-17 at 6 05 14 PM

- [x] update icons (attached Zip of icon SVGs below)
- search icon
- (X) close icon

**Feedback for statement pagination #41568**

- [ ] Resolve scrolling issue for page 2: page jumps to the top when going from page 1 to page 2, but then stays scrolls to bottom for the rest of the pages.
StatementsPage_1

- [ ] update icons - attached Zip of icon SVGs below, if these don't look right we can keep the current icons
- (<) previous icon
- (>) next icon
- (...) ellipsis icon

[SVG-icons.zip](https://github.com/cockroachdb/cockroach/files/3805370/SVG-icons.zip)

### PR DESCRIPTION
**Pull request by vladlos**
statements page redesign, because of table component is reused on other pages, it changed view there also.

Resolves: #42154 

Release note (ui): none

Screenshot 2020-01-10 at 17 56 32
Screenshot 2020-01-10 at 17 56 40

### CODE REVIEW

**Comment by vilterp** [comment]
This may be an @Annebirzin question, but why put the logical plan and stats on different tabs? Just seems like more clicks to get the same information.

**Comment by Annebirzin** [comment]
@vilterp the main thinking there was for scalability. It seemed that the statement schema or logical plans could get very long in which case showing everything in one view would require lots of scrolling (not to mention the logical plan might get lost below the fold).

But if very long statement schemas or logical plans are an edge case then we could consider keeping them in one tab. We can also test this version out with users to see if they are able to access logical plans intuitively.

**Review by dhartunian — CHANGES_REQUESTED** [review]
> Reviewed 14 of 36 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43875)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster)

___
*[pkg/ui/src/views/shared/components/summaryCard/imdex.tsx, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_GsmoD7EmAvY0Ec6R:-Ly_GsmoD7EmAvY0Ec6S:b-nm72fz) ([raw file](https://github.com/cockroachdb/cockroach/blob/827e768a309a1aa3942326d3bc4711426f3464d4/pkg/ui/src/views/shared/components/summaryCard/imdex.tsx#L1)):*
> ```XML
> // Copyright 2018 The Cockroach Authors.
> ```

@elkmaster Please rename this file to `index.tsx` and change the license to match the other javascript files.

**Review by dhartunian — CHANGES_REQUESTED** [review]
As discussed offline with @elkmaster, we will be breaking this PR up into smaller pieces.

> Reviewed 2 of 36 files at r1, 7 of 7 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-:-Lya-S-W7OSXM1xhxLgG:bs7to8t)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster)

___
*[pkg/ui/src/views/app/components/empty/index.tsx, line 34 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_O9sQ0zEFzoSx0x_4:-Ly_O9sQ0zEFzoSx0x_5:bkt485f) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/app/components/empty/index.tsx#L34)):*
> ```XML
>       <h2 className="empty-container__title">{title}</h2>
>       <p className="empty-container__description">{description}</p>
>       <Button className="empty-container__button" type="primary">{button}</Button>
> ```

This button doesn't link to anything. Am I misunderstanding its purpose?
___
*[pkg/ui/src/views/app/components/empty/styles.styl, line 48 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_Zsgc8R3scwRnuIwh:-Ly_Zsgc8R3scwRnuIwi:b-8h9vim) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/app/components/empty/styles.styl#L48)):*
> ```Stylus
>       background-color $green
>       opacity 0.7
>       color $adminui-white !important
> ```

we should avoid !important. Why is this necessary?
___
*[pkg/ui/src/views/app/components/Search/index.tsx, line 11 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly__EO12FkWJBLXdaiJ:-Ly__EO12FkWJBLXdaiK:b-brrcms) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/app/components/Search/index.tsx#L11)):*
> ```XML
> // licenses/APL.txt.
> 
> import { Button, Form, Icon, Input } from "antd";
> ```

Icon is no longer used, please remove from the imports
___
*[pkg/ui/src/views/shared/components/sql/box.tsx, line 24 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_ZBQC3EIqy3SUJ8N6:-Ly_ZBQD2Wz4uA4lTsEN:bmt2q04) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/shared/components/sql/box.tsx#L24)):*
> ```XML
> }
> 
> const regExpTables = /(?<=(FROM|INTO|TABLE|UPDATE|CREATE)\s)([^\s([]+)/g;
> ```

I'll look into whether the backend can tell you what the table name is.

FYI Lookbehind regexp syntax is not supported in Firefox yet, avoid using it for now.
https://github.com/sindresorhus/ip-regex/issues/24
___
*[pkg/ui/src/views/shared/components/summaryBar/index.tsx, line 61 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_oTN_BOh-uqewMfF1:-Ly_oTNa2vwAt910A62l:bfdtpdc) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/shared/components/summaryBar/index.tsx#L61)):*
> ```XML
> }
> 
> export function computeValue(value: number, format: (n: number) => string = numberToString) {
> ```

Since this function will be used outside this module, can you give it a more descriptive name? Maybe `formatNumberForDisplay`
___
*[pkg/ui/src/views/statements/barCharts.tsx, line 209 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/43875#-Ly_xJNI0mnm5nnLJk0X:-Ly_xJNO4Ei37_LhSJDo:bilwo1n) ([raw file](https://github.com/cockroachdb/cockroach/blob/1de22ef361b33a27b48f8e865c2337b11b1cc227/pkg/ui/src/views/statements/barCharts.tsx#L209)):*
> ```XML
>   return {
>     firstAttemptsBarChart() {
>       return firstAttempts;
> ```

These values being returned are no longer bar charts so you either need to make a new file, or restructure the naming in this one.

**Review by dhartunian — CHANGES_REQUESTED** [review]
Thanks for restructuring this @elkmaster. Let me know if you have any questions about my comments.

File: pkg/ui/src/views/statements/barCharts.tsx [hunk 1]
```diff
@@ -193,57 +200,25 @@ export function countBreakdown(s: StatementStatistics) {
   const retries = count - firstAttempts;
   const maxRetries = longToInt(s.stats.max_retries);
 
-  const scale = d3.scale.linear()
-    .domain([0, count])
-    .range([0, 100]);
+  // const scale = d3.scale.linear()
+  //   .domain([0, count])
+  //   .range([0, 100]);
```
[the remark this answers is not in the document]
Comment (elkmaster) [reply pkg/ui/src/views/statements/barCharts.tsx]: this change is outdated. in latest commit it's not commented.

File: pkg/ui/src/views/shared/components/sortabletable/index.tsx [hunk 2]
```diff
@@ -68,6 +68,7 @@ interface TableProps {
   // a full-width area below it when expanded.
   expandableConfig?: ExpandableConfig;
   drawer?: boolean;
+  firstCellHeader?: boolean;
```
[the remark this answers is not in the document]
Comment (elkmaster) [reply pkg/ui/src/views/shared/components/sortabletable/index.tsx]: this prop is needed to add border-right for first cell of the row like here:
Снимок экрана 2020-01-28 в 10 25 15

I'll rename prop to be more clear

**Review by dhartunian — APPROVED** [review]

**Pull request merged**